src/diffusers/schedulers/scheduling_dpmsolver_multistep.py

```diff
@@ -407,6 +409,11 @@ def set_timesteps(
             sigmas = np.flip(sigmas).copy()
             sigmas = self._convert_to_beta(in_sigmas=sigmas, num_inference_steps=num_inference_steps)
             timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas])
+        elif self.config.use_flow_sigmas:
```

Comment:
```suggestion
        elif self.config.use_beta_sigmas:
            sigmas = np.flip(sigmas).copy()
            sigmas = self._convert_to_beta(in_sigmas=sigmas, num_inference_steps=num_inference_steps)
            timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas])
            if self.config.use_flow_sigmas:
                alphas = np.linspace(1, 1 / self.config.num_train_timesteps, num_inference_steps + 1)
                sigmas = 1.0 - alphas
                sigmas = np.flip(self.config.flow_shift * sigmas / (1 + (self.config.flow_shift - 1) * sigmas))[:-1]
                timesteps = (sigmas * self.config.num_train_timesteps).copy()
```
Do you mean the logic like this, but with some change of code?

Reply:
umm not sure what you mean, here in your suggested code change you have this, 


```python
if self.config.use_beta_sigmas
    if self.config.use_flow_sigmas:
        ....
```

I don't think we can configure `use_beta_sigma` and `use_flow_sigmas` to be True at the same time. However, we should be able to configure `use_beta_sigma=True` and `prediction_type="flow_prediction"` at the same time, no? basically, you will still be doing flow match but use a "noise schedule" that's not uniform

Reply:
I just wonder what happens if the user does that with the current implementation. Would it work?

Reply:
> you will still be doing flow match but use a "noise schedule" that's not uniform

Then, it will not work I think, the noise schedule has to be uniform as it's defined in SD3, Flux and Sana.
If I understand correct, the code will work, like:

```python
if self.config.use_beta_sigmas
    if prediction_type=="flow_prediction":
        ....
```
But, I'm not sure if the sigma here is possible to be uniform here:
https://github.com/huggingface/diffusers/blob/29e93b7116046490be05bf571a5b71600f79df27/src/diffusers/schedulers/scheduling_dpmsolver_multistep.py#L596-L605
If the sigma can't be uniform, I think it's not reasonable to place `prediction_type="flow_prediction"` under `use_beta_sigma=True`

Reply:
ohh I was under the impression that we can use other sigmas distribution for flow-matching see https://github.com/huggingface/diffusers/pull/10001#issuecomment-2501839522

but we don't have to worry about it for this PR.  If you have time and are interested in investigating this it would be great! :)  if not, we can just make sure user are only be able use `use_flow_sigma=True` when `prediction_type="flow_prediction"`, i.e. throw an error if not the case

Reply:
> ohh I was under the impression that we can use other sigmas distribution for flow-matching see [#10001 (comment)](https://github.com/huggingface/diffusers/pull/10001#issuecomment-2501839522)
> 
> but we don't have to worry about it for this PR. If you have time and are interested in investigating this it would be great! :) if not, we can just make sure user are only be able use `use_flow_sigma=True` when `prediction_type="flow_prediction"`, i.e. throw an error if not the case

Interesting, and it makes sense. I had a experiment before, If I train the model with timestep_shift=3 and usetimestep_shift=4 to inference will also work well. This may explain why the sigmas change a little bit, but the model can still work, specially for the large model, like FLUX.